code/game/machinery/dance_machine.dm

```diff
@@ -79,49 +74,40 @@
 			return
 	return ..()
 
-/obj/machinery/disco/update_icon()
+/obj/machinery/jukebox/update_icon()
 	if(active)
-		icon_state = "disco1"
+		icon_state = "[initial(icon_state)]-active"
 	else
-		icon_state = "disco0"
-	..()
-
+		icon_state = "[initial(icon_state)]"
 
-/obj/machinery/disco/ui_interact(mob/user)
+/obj/machinery/jukebox/ui_interact(mob/user)
 	. = ..()
 	if (!anchored)
 		to_chat(user,"<span class='warning'>This device must be anchored by a wrench!</span>")
 		return
 	if(!allowed(user))
-		to_chat(user,"<span class='warning'>Error: Access Denied - Message: Only the engineering department can be trusted with this kind of power.</span>")
+		to_chat(user,"<span class='warning'>Error: Access Denied.</span>")
 		user.playsound_local(src,'sound/misc/compiler-failure.ogg', 25, 1)
 		return
 	if(!Adjacent(user) && !isAI(user))
 		return
+	if(!songs.len)
+		to_chat(user,"<span class='warning'>Error: No music tracks have been authorized for your station. Petition Central Command to resolve this issue.</span>")
+		user.playsound_local(src,'sound/misc/compiler-failure.ogg', 25, 1)
```

Comment:
just playsound should be fine

Reply:
@Robustin Was it on purpose that only the user will recieve the error noise?